### PR TITLE
Deploy guide for Rancher/RKE

### PR DESCRIPTION
**Pull request by tapih**
https://github.com/cybozu-go/topolvm/issues/91

### CODE REVIEW

File: docs/rancher.md [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+### Apply scheduler extension
+
+Download the scheduler extension configuration files onto `master`.
+
+```bash
+gcloud compute ssh --zone ${ZONE} master
+
+TOPOLVM_VERSION=0.4.0
+sudo mkdir -p /etc/kubernetes/scheduler
+sudo curl -sSL -o /etc/kubernetes/scheduler/scheduler-config.yaml https://raw.githubusercontent.com/masa213f/example-topolvm/master/scheduler-config/scheduler-config.yaml
```
Comment (ymmt2005) [inline docs/rancher.md]: Do not reference `masa213f`.

File: docs/rancher.md [hunk 2]
```diff
@@ -197,7 +197,7 @@ First, `master` has the following label and taint.
 
 To locate `topolvm-scheduler` onto `master`, update node affinity and toleration.
 
-```bash
+```diff
```
Comment (ymmt2005) [inline docs/rancher.md]: It's not a diff.

File: docs/rancher.md [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+- Label
+    1. `node-role.kubernetes.io/controlplane=true`
+    2. `node-role.kubernetes.io/etcd=true`
+- Taints
+    1. `node-role.kubernetes.io/controlplane=true:NoSchedule`
+    2. `node-role.kubernetes.io/etcd=true:NoExecute`
+
+To locate `topolvm-scheduler` onto `master`, update node affinity and toleration.
+
+```bash
```
Comment (ymmt2005) [inline docs/rancher.md]: Please replace all `bash` occurrences with `console` as they all look like console I/O.

File: docs/rancher.md [hunk 4]
```diff
@@ -0,0 +1,320 @@
+How to use TopoLVM with Rancher/RKE
+===================================
+
+This document is a brief introduction of how to use TopoLVM on [Rancher/RKE](https://rancher.com/docs/rke/latest/en/).
```
Comment (masa213f) [inline docs/rancher.md]: "gcloud" command (and initialize the command) is required to try this guide.
So please write the requirement.

File: docs/rancher.md [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+## 1. Run Rancher Server
+
+### Create GCP instance
+
+Create a GCP instance for Rancher Server.
+
+```bash
+ZONE=asia-northeast1-c
+gcloud compute instances create rancher \
+  --zone ${ZONE} \
```
Comment (masa213f) [inline docs/rancher.md]: Can I omit the `--zone` option from `gcloud` if the `gcloud` command is configured properly?

File: docs/rancher.md [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
++      - key: node-role.kubernetes.io/controlplane
++        operator: Exists
++      - key: node-role.kubernetes.io/etcd
++        operator: Exists
+...
+```
+
+### Apply scheduler extension
+
+Download the scheduler extension configuration files onto `master`.
```
Comment (masa213f) [inline docs/rancher.md]: Please write that we need to put the config file under `/etc/kubernetes` and that reason.

File: docs/rancher.md (hunk 5 above)
Comment (tapih) [reply docs/rancher.md]: I think so but I'm not sure if I should change the commands as you said.

Comment (masa213f) [reply docs/rancher.md]: OK. Please try and fix appropriately.

Comment (tapih) [reply docs/rancher.md]: Why do you think that should be changed?
There is only a little difference, but I prefer not to force users to update their settings(=zone).

Comment (masa213f) [reply docs/rancher.md]: Why do you force users to use `asia-northeast1-c`?
This is an official guide. If the default zone is set properly, we should let users use the zone which the user wants, I think.

Comment (tapih) [reply docs/rancher.md]: I don't mean to force them to use `asia-northeast1-c`.
Which zone they use is optional, but I just wanted to give it with `--zone`.

Comment (masa213f) [reply docs/rancher.md]: I don't say anything anymore.

Of course, my base documentation is bad too.
But I think it's not good as an official document.

File: docs/rancher.md [hunk 7]
```diff
@@ -0,0 +1,324 @@
+How to use TopoLVM with Rancher/RKE
+===================================
+
+This document is a brief introduction of how to use TopoLVM on [Rancher/RKE](https://rancher.com/docs/rke/latest/en/).
+
+4 GCP instances will be used in this example. Each instance will be configured to have the following role.
```
Comment (ymmt2005) [inline docs/rancher.md]:
```suggestion
Rancher/RKE will be deployed on the following 4 instances of Google Compute Engine (GCE).
```

File: docs/rancher.md [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+gcloud compute ssh --zone ${ZONE} master -- "curl -sSLf https://get.docker.com | sudo sh"
+gcloud compute ssh --zone ${ZONE} worker1 -- "curl -sSLf https://get.docker.com | sudo sh"
+gcloud compute ssh --zone ${ZONE} worker2 -- "curl -sSLf https://get.docker.com | sudo sh"
+```
+
+### Deploy Kubernetes components with Rancher
+
+Go to the Rancher dashboard and click `Add Cluster` -> `From existing nodes (Custom)`.
+
+Then configuration page will come up. Overwrite some default values with the following.
```
Comment (ymmt2005) [inline docs/rancher.md]:
```suggestion
Go to the Rancher dashboard and click `Add Cluster` -> `From existing nodes (Custom)`
to see the configuration page.  Overwrite some default values as follows.
```

File: docs/rancher.md [hunk 9]
```diff
@@ -0,0 +1,324 @@
...
+-      - effect: NoSchedule
+-        key: node-role.kubernetes.io/master
++      - key: node-role.kubernetes.io/controlplane
++        operator: Exists
++      - key: node-role.kubernetes.io/etcd
++        operator: Exists
+...
+```
+
+### Apply scheduler extension
```
Comment (ymmt2005) [inline docs/rancher.md]:
```suggestion
### Adding the scheduler extender
```

File: docs/rancher.md [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+
+Download the scheduler extension configuration files onto `master`.
```
Comment (ymmt2005) [inline docs/rancher.md]:
```suggestion
Download the scheduler extender configuration files on the `master` instance.
```

File: docs/rancher.md [hunk 11]
```diff
@@ -0,0 +1,324 @@
...
+```console
+TOPOLVM_VERSION=0.4.0
+kubectl apply -k https://github.com/cybozu-go/topolvm/deploy/manifests?ref=v${TOPOLVM_VERSION}
+kubectl apply -f https://raw.githubusercontent.com/cybozu-go/topolvm/v${TOPOLVM_VERSION}/deploy/manifests/certificates.yaml
+```
+
+## 6. Configure `topolvm-scheduler`
+
+You almost finish deploying TopoLVM here, but you need a little tweak to make `topolvm-scheduler` work properly.
+
```
Comment (ymmt2005) [inline docs/rancher.md]: Seems spurious.

```suggestion
```

File: docs/rancher.md [hunk 12]
```diff
@@ -0,0 +1,324 @@
...
+   ssh_agent_auth: false
+```
+
+Then click `Save` to finish the configuration.
+
+## 7. Provision volume with TopoLVM
+
+Congratulations!! You finally deployed TopoLVM on RKE.
+
+To confirm now TopoLVM is actually working, create PVC and mount it on a `Pod`.
```
Comment (ymmt2005) [inline docs/rancher.md]:
```suggestion
To confirm TopoLVM is working, create PVC and mount it on a `Pod`.
```

**Pull request merged**